src/blob.jl

```diff
@@ -59,32 +59,30 @@ Defaults to `sizeof(T)`.
     if isempty(fieldnames(T))
         quote
             $(Expr(:meta, :inline))
-            sizeof(T)
+            $(sizeof(T))
```

Comment:
Why is this better?

Reply:
If you don't put `$(...)` then an actual `sizeof(T)` call happens at runtime. This way, it's computed once at code generation time and a constant is returned at runtime.

Reply:
Are you sure?

`julia> foo(::T) where T = sizeof(T)
foo (generic function with 1 method)

julia> struct Bar x::Float64 end

julia> foo(Bar(1.1))
8

julia> @code_native foo(Bar(1.1))
	.section	__TEXT,__text,regular,pure_instructions
; Function foo {
; Location: REPL[13]:1
	movl	$8, %eax
	retl
	nopw	%cs:(%eax,%eax)
;}

julia> @code_native foo(12)
	.section	__TEXT,__text,regular,pure_instructions
; Function foo {
; Location: REPL[13]:1
	movl	$8, %eax
	retl
	nopw	%cs:(%eax,%eax)
;}`

Reply:
It's probably a compiler optimization that partially evaluates simple function, but like all compiler optimization, there's no guarantee for that to kick-in. Though, by putting `$(...)` around `sizeof(T)` call, it's guaranteed to be a constant. Evidently, it won't become a constant in more complicated code like below.